docs/discussion/concepts/common_unit.md

```diff
@@ -0,0 +1,175 @@
+    - This should _rarely, if ever_ be named in code.
+        - In implementations, we need to do this, for example, for defining the _unit label_ of
+          a `CommonUnit<...>`, or defining its ordering relative to other units.
+        - In end user code, this should probably _never_ be named.
+        - In either case: **never** write `CommonUnit<...>` with _specific_ template arguments!
+          Only use it for matching.
+    - So then, how can `CommonUnit<...>` arise?  Only as _the result of some type computation_.
+- **`CommonUnitT<...>`**.  This _computes_ the common unit of the provided units.
+
+Let's clarify this relationship with an example.  Suppose you're writing a function based on two
+arbitrary (but same-Dimension) units, `U1` and `U2`, and you need their "common unit".
+
+- What you would _write_ is `CommonUnitT<U1, U2>`, **not** `CommonUnit<U1, U2>`.
+    - `CommonUnitT<...>` says "_please calculate_ the common unit".
+    - `CommonUnit<...>` says "_this is the result_ of calculating the common unit".
+- What you _get_ depends on the specific units.
+    - For `CommonUnitT<Inches, Meters>`, the result might be `CommonUnit<Inches, Meters>`.[^1]  This
+      is because the greatest common divisor for `Inches` and `Meters` is smaller than both of them.
+    - For `CommonUnitT<Inches, Feet>`, the result would simply be `Inches`, because `Inches` is
+      quantity-equivalent to this common unit (it evenly divides both `Inches` and `Feet`).
+
+[^1]:  It might also be `CommonUnit<Meters, Inches>`.  The ordering is deterministic, but
+unspecified.
+
+??? info "Implementation approach details (deep in the weeds)"
+
+    There are two main tools we use to implement `CommonUnitT`.
+
+    1. `FlatDedupedTypeList`.  For a given variadic pack `List<...>` (which, for us, will be
+       `CommonUnit<...>`), `FlatDedupedTypeList<List, Ts...>` will produce a `List<...>`, whose
+       elements are `Ts...`, but sorted according to `InOrderFor<List, ...>`, and with duplicates
+       removed.
+
+        - If any of the `Ts` are already `List<Us...>`, we effectively "unpack" it, replacing it with
+          `Us...`.  This is the "flat" part in `FlatDedupedTypeList`.
+
+    2. `FirstQuantityEquivalentUnit`.  The above step produces a `CommonUnit<...>` specialization, which
+       itself meets [the definition of a unit](../../reference/unit.md).  But is it the unit we
+       really want to provide?  Not if there's a simpler one!
+       `FirstQuantityEquivalentUnit<CommonUnit<Us...>>` searches through the unit list `Us...`, and
+       returns the first quantity-equivalent one it finds. If no such unit is available, then we
+       fall back to returning `CommonUnit<Us...>`.
+
+## Changes for `QuantityPoint` {#common-quantity-point}
+
+The common unit for `QuantityPoint` operations is different from the common unit for `Quantity`.  To see why
+a single notion of "common unit" isn't enough, consider `Celsius` and `Kelvins`.
```

Comment:
Pedantry, perhaps, but both "Kelvin" and "Kelvins" are valid stylistic choices (e.g. Oxford prefers plural "Kelvin", Merriam-Webster prefers plural "Kelvins"). Is this going to become relevant for users if their tendency has been to use one vs. the other?

Reply:
Acknowledged!